### DISCUSSION TITLE
Ability to open layout (json file) directly in app

### DISCUSSION THREAD

**Discussion by jtbandes** [foxglove/community#311] [Q&A]
For parity with `rviz -d config.rviz`, it would be nice to be able to launch Studio with `foxglove-studio -l layout.json` or similar.

Related: foxglove/community#192 & foxglove/studio#2117 (loading layouts from remote/hosted sources)

**Comment by defunctzombie** [comment]
Is the idea here that you would keep a layout file in your repo? Do you think this would also update layout changes _back_ to that file? What would this do if you are signed in?

**Comment by jtbandes** [comment]
It would be especially useful for robotics toolkits/projects that want to distribute an example layout. The simplest behavior would be to just create a new layout each time it's invoked this way. What rviz does is treat the current layout as unsaved and prompts you to save when you try to open a different layout or close the app.
